AngularJS/src/org/angular2/entities/source/Angular2SourceDirective.java

```diff
@@ -114,6 +122,16 @@ public boolean isTemplate() {
     );
   }
 
+  @NotNull
+  private Angular2CtorBoundParameters getCachedCtorParameters() {
```

Comment:
Please incorporate into `getProperties()` method to avoid traversing class properties twice.